fedimint-atomic-broadcast/src/broadcast.rs

```diff
@@ -0,0 +1,148 @@
+use std::time::Duration;
+
+use fedimint_core::db::Database;
+use fedimint_core::PeerId;
+use tokio::sync::{mpsc, oneshot, watch};
+
+use crate::keychain::Keychain;
+use crate::{
+    db, session, AcceptedItem, Decision, Message, OrderedItem, Recipient, Shutdown, SignedBlock,
+};
+
+async fn relay_messages(
+    db: Database,
+    incoming_message_receiver: async_channel::Receiver<(Message, PeerId)>,
+    outgoing_message_sender: async_channel::Sender<(Message, Recipient)>,
+    network_data_sender: async_channel::Sender<Vec<u8>>,
+    signed_block_sender: async_channel::Sender<SignedBlock>,
+) {
+    while let Ok((message, peer_id)) = incoming_message_receiver.recv().await {
+        match message {
+            Message::NetworkData(network_data) => {
+                network_data_sender.send(network_data).await.ok();
```

Comment:
instead of `ok()` should we `expect()` since the channel should never close?

Reply:
It is correct that the network data channel and signed block channel actually should never be closed. However, I did generally avoid to panic if there is no irreparable corruption of the systems state. If we use expect and refactor at some point we might overlook it and could get a race condition.